src/components/form/index.tsx

```diff
@@ -118,7 +118,9 @@ export class SmoothlyForm implements ComponentWillLoad, Clearable, Submittable,
 		if (Input.Element.is(event.target)) {
 			if (await event.target.binary?.())
 				this.contentType = "form-data"
-			this.value = Data.merge(this.value, { [event.target.name]: await event.target.getValue() })
+			const inputValue = await event.target.getValue()
+			this.value = Data.merge(this.value, { [event.target.name]: inputValue })
+			this.smoothlyFormInput.emit(Data.convertArrays(this.value))
```

Comment:
are you sure that it is not this line that fixes your issue? the two other lines can be merged. it should not matter that you break it out to a variable.

Reply:
I did not think it would work either but it very clearly does.


These are both test with just reloading the page.

### With race-condition
![image](https://github.com/user-attachments/assets/f9d259c6-7425-4ec4-915c-fb7181d2725b)


### Without race-condition
![image](https://github.com/user-attachments/assets/135c284d-10a4-4803-87d3-0aa7e6f3c69e)

Reply:
:+1: